client/src/modules/user/AuthBox/components/PasswordChangeForm.jsx

```diff
@@ -0,0 +1,296 @@
+import React, { Fragment, PureComponent } from 'react';
+import { FormattedMessage, injectIntl } from 'react-intl';
+import PropTypes from 'prop-types';
+import _some from 'lodash/some';
+import validator from 'validator';
+
+import AuthInput from './AuthInput';
+import { changePassword } from 'modules/user/model/actions';
+import { AbortPromiseException } from 'common/exceptions/AbortPromiseException';
+import { makeAbortablePromise } from 'common/utils/helpers';
+import PendingButton from 'common/components/PendingButton';
+import { IntlPropType } from 'common/constants/propTypes';
+import { ValidationException } from 'common/exceptions/ValidationException';
+
+class PasswordChangeForm extends PureComponent {
+  pendingPromise = null;
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      confirmPasswordValue: undefined,
+      higherLevelErrors: {
+        confirmPasswordValueError: '',
+        newPasswordError: ''
+      },
+      isFormValid: false,
+      isNewPasswordValid: false,
+      isPasswordConfirmValid: false,
+      isPasswordValid: false,
+      newPassword: '',
+      password: '',
+      passwordChanged: false,
+      pending: false,
+      changePasswordErrorId: ''
+    };
+  }
+
+  componentDidUpdate() {
+    this.isFormValid();
+  }
+
+  componentWillUnmount() {
+    if (this.pendingPromise) {
+      this.pendingPromise.abort();
+    }
+  }
+
+  onPasswordChange = (password, isValid) => {
+    const { changePasswordErrorId } = this.state;
+    const error = isValid ? '' : changePasswordErrorId;
+
+    this.setState(
+      {
+        password,
+        isPasswordValid: isValid && password.length > 0,
+        changePasswordErrorId: error
+      },
+      this.comparePasswords
+    );
+  };
+
+  onNewPasswordChange = (newPassword, isValid) => {
+    const {
+      higherLevelErrors,
+      higherLevelErrors: { newPasswordError }
+    } = this.state;
+    const error = isValid ? '' : newPasswordError;
+
+    this.setState(
+      {
+        higherLevelErrors: {
+          ...higherLevelErrors,
+          newPasswordError: error
+        },
+        newPassword,
+        isNewPasswordValid: isValid
+      },
+      this.comparePasswords
+    );
+  };
+
+  onPasswordConfirmChange = (confirmPasswordValue, isValid) =>
+    this.setState(
+      { confirmPasswordValue, isPasswordConfirmValid: isValid },
+      this.comparePasswords
+    );
+
+  passwordValidator = value => {
+    const { matches } = validator;
+
+    if (!matches(value, /^[^\s]{4,32}$/)) {
```

Comment:
this should be put in some common function as we have same check during password recovery